### ISSUE TITLE
Odd loop is wrongly optimized.

### ISSUE THREAD

**Issue by Themaister** [KhronosGroup/SPIRV-Tools#989]
I have this (very reduced) test shader in SPIRV-Cross:
```
#version 310 es
layout(local_size_x = 1) in;

layout(std430, binding = 0) readonly buffer SSBO
{
    mat4 mvp;
    vec4 in_data[];
};

layout(std430, binding = 1) writeonly buffer SSBO2
{
    vec4 out_data[];
};

void main()
{
    int k = 0;
	uint i = 0u;
    uint ident = gl_GlobalInvocationID.x;
	for (;;)
	{
		i += 2u;
		if (i > 20u)
			break;
	}

    vec4 idat = in_data[ident];
    out_data[ident] = idat;
}
```

When I compile that with glslang I get:
```
; SPIR-V
; Version: 1.0
; Generator: Khronos Glslang Reference Front End; 1
; Bound: 56
; Schema: 0
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint GLCompute %main "main" %gl_GlobalInvocationID
               OpExecutionMode %main LocalSize 1 1 1
               OpSource ESSL 310
               OpName %main "main"
               OpName %k "k"
               OpName %i "i"
               OpName %ident "ident"
               OpName %gl_GlobalInvocationID "gl_GlobalInvocationID"
               OpName %idat "idat"
               OpName %SSBO "SSBO"
               OpMemberName %SSBO 0 "mvp"
               OpMemberName %SSBO 1 "in_data"
               OpName %_ ""
               OpName %SSBO2 "SSBO2"
               OpMemberName %SSBO2 0 "out_data"
               OpName %__0 ""
               OpDecorate %gl_GlobalInvocationID BuiltIn GlobalInvocationId
               OpDecorate %_runtimearr_v4float ArrayStride 16
               OpMemberDecorate %SSBO 0 ColMajor
               OpMemberDecorate %SSBO 0 NonWritable
               OpMemberDecorate %SSBO 0 Offset 0
               OpMemberDecorate %SSBO 0 MatrixStride 16
               OpMemberDecorate %SSBO 1 NonWritable
               OpMemberDecorate %SSBO 1 Offset 64
               OpDecorate %SSBO BufferBlock
               OpDecorate %_ DescriptorSet 0
               OpDecorate %_ Binding 0
               OpDecorate %_runtimearr_v4float_0 ArrayStride 16
               OpMemberDecorate %SSBO2 0 NonReadable
               OpMemberDecorate %SSBO2 0 Offset 0
               OpDecorate %SSBO2 BufferBlock
               OpDecorate %__0 DescriptorSet 0
               OpDecorate %__0 Binding 1
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
        %int = OpTypeInt 32 1
%_ptr_Function_int = OpTypePointer Function %int
      %int_0 = OpConstant %int 0
       %uint = OpTypeInt 32 0
%_ptr_Function_uint = OpTypePointer Function %uint
     %uint_0 = OpConstant %uint 0
     %v3uint = OpTypeVector %uint 3
%_ptr_Input_v3uint = OpTypePointer Input %v3uint
%gl_GlobalInvocationID = OpVariable %_ptr_Input_v3uint Input
%_ptr_Input_uint = OpTypePointer Input %uint
     %uint_2 = OpConstant %uint 2
    %uint_20 = OpConstant %uint 20
       %bool = OpTypeBool
      %float = OpTypeFloat 32
    %v4float = OpTypeVector %float 4
%_ptr_Function_v4float = OpTypePointer Function %v4float
%mat4v4float = OpTypeMatrix %v4float 4
%_runtimearr_v4float = OpTypeRuntimeArray %v4float
       %SSBO = OpTypeStruct %mat4v4float %_runtimearr_v4float
%_ptr_Uniform_SSBO = OpTypePointer Uniform %SSBO
          %_ = OpVariable %_ptr_Uniform_SSBO Uniform
      %int_1 = OpConstant %int 1
%_ptr_Uniform_v4float = OpTypePointer Uniform %v4float
%_runtimearr_v4float_0 = OpTypeRuntimeArray %v4float
      %SSBO2 = OpTypeStruct %_runtimearr_v4float_0
%_ptr_Uniform_SSBO2 = OpTypePointer Uniform %SSBO2
        %__0 = OpVariable %_ptr_Uniform_SSBO2 Uniform
       %main = OpFunction %void None %3
          %5 = OpLabel
          %k = OpVariable %_ptr_Function_int Function
          %i = OpVariable %_ptr_Function_uint Function
      %ident = OpVariable %_ptr_Function_uint Function
       %idat = OpVariable %_ptr_Function_v4float Function
               OpStore %k %int_0
               OpStore %i %uint_0
         %19 = OpAccessChain %_ptr_Input_uint %gl_GlobalInvocationID %uint_0
         %20 = OpLoad %uint %19
               OpStore %ident %20
               OpBranch %21
         %21 = OpLabel
               OpLoopMerge %23 %24 None
               OpBranch %22
         %22 = OpLabel
         %26 = OpLoad %uint %i
         %27 = OpIAdd %uint %26 %uint_2
               OpStore %i %27
         %28 = OpLoad %uint %i
         %31 = OpUGreaterThan %bool %28 %uint_20
               OpSelectionMerge %33 None
               OpBranchConditional %31 %32 %33
         %32 = OpLabel
               OpBranch %23
         %33 = OpLabel
               OpBranch %24
         %24 = OpLabel
               OpBranch %21
         %23 = OpLabel
         %45 = OpLoad %uint %ident
         %47 = OpAccessChain %_ptr_Uniform_v4float %_ %int_1 %45
         %48 = OpLoad %v4float %47
               OpStore %idat %48
         %53 = OpLoad %uint %ident
         %54 = OpLoad %v4float %idat
         %55 = OpAccessChain %_ptr_Uniform_v4float %__0 %int_0 %53
               OpStore %55 %54
               OpReturn
               OpFunctionEnd
```

When I optimize with -O, I get:
```
; SPIR-V
; Version: 1.0
; Generator: Khronos Glslang Reference Front End; 1
; Bound: 56
; Schema: 0
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint GLCompute %main "main" %gl_GlobalInvocationID
               OpExecutionMode %main LocalSize 1 1 1
               OpSource ESSL 310
               OpName %main "main"
               OpName %gl_GlobalInvocationID "gl_GlobalInvocationID"
               OpName %SSBO "SSBO"
               OpMemberName %SSBO 0 "mvp"
               OpMemberName %SSBO 1 "in_data"
               OpName %_ ""
               OpName %SSBO2 "SSBO2"
               OpMemberName %SSBO2 0 "out_data"
               OpName %__0 ""
               OpDecorate %gl_GlobalInvocationID BuiltIn GlobalInvocationId
               OpDecorate %_runtimearr_v4float ArrayStride 16
               OpMemberDecorate %SSBO 0 ColMajor
               OpMemberDecorate %SSBO 0 NonWritable
               OpMemberDecorate %SSBO 0 Offset 0
               OpMemberDecorate %SSBO 0 MatrixStride 16
               OpMemberDecorate %SSBO 1 NonWritable
               OpMemberDecorate %SSBO 1 Offset 64
               OpDecorate %SSBO BufferBlock
               OpDecorate %_ DescriptorSet 0
               OpDecorate %_ Binding 0
               OpDecorate %_runtimearr_v4float_0 ArrayStride 16
               OpMemberDecorate %SSBO2 0 NonReadable
               OpMemberDecorate %SSBO2 0 Offset 0
               OpDecorate %SSBO2 BufferBlock
               OpDecorate %__0 DescriptorSet 0
               OpDecorate %__0 Binding 1
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
        %int = OpTypeInt 32 1
%_ptr_Function_int = OpTypePointer Function %int
      %int_0 = OpConstant %int 0
       %uint = OpTypeInt 32 0
%_ptr_Function_uint = OpTypePointer Function %uint
     %uint_0 = OpConstant %uint 0
     %v3uint = OpTypeVector %uint 3
%_ptr_Input_v3uint = OpTypePointer Input %v3uint
%gl_GlobalInvocationID = OpVariable %_ptr_Input_v3uint Input
%_ptr_Input_uint = OpTypePointer Input %uint
     %uint_2 = OpConstant %uint 2
    %uint_20 = OpConstant %uint 20
       %bool = OpTypeBool
      %float = OpTypeFloat 32
    %v4float = OpTypeVector %float 4
%_ptr_Function_v4float = OpTypePointer Function %v4float
%mat4v4float = OpTypeMatrix %v4float 4
%_runtimearr_v4float = OpTypeRuntimeArray %v4float
       %SSBO = OpTypeStruct %mat4v4float %_runtimearr_v4float
%_ptr_Uniform_SSBO = OpTypePointer Uniform %SSBO
          %_ = OpVariable %_ptr_Uniform_SSBO Uniform
      %int_1 = OpConstant %int 1
%_ptr_Uniform_v4float = OpTypePointer Uniform %v4float
%_runtimearr_v4float_0 = OpTypeRuntimeArray %v4float
      %SSBO2 = OpTypeStruct %_runtimearr_v4float_0
%_ptr_Uniform_SSBO2 = OpTypePointer Uniform %SSBO2
        %__0 = OpVariable %_ptr_Uniform_SSBO2 Uniform
       %main = OpFunction %void None %3
          %5 = OpLabel
         %19 = OpAccessChain %_ptr_Input_uint %gl_GlobalInvocationID %uint_0
         %20 = OpLoad %uint %19
               OpBranch %21
         %21 = OpLabel
               OpLoopMerge %23 %24 None
               OpBranch %22
         %22 = OpLabel
               OpBranch %24
         %24 = OpLabel
               OpBranch %21
         %23 = OpLabel
         %47 = OpAccessChain %_ptr_Uniform_v4float %_ %int_1 %20
         %48 = OpLoad %v4float %47
         %55 = OpAccessChain %_ptr_Uniform_v4float %__0 %int_0 %20
               OpStore %55 %48
               OpReturn
               OpFunctionEnd
```
The loop never terminates.

**Comment by greg-lunarg** [comment]
Thanks! Will take a look.

**Comment by dneto0** [comment]
Fixed by #1030
